code/modules/awaymissions/mission_code/vrhub.dm

```diff
@@ -0,0 +1,169 @@
+		if(!findspell)
+			user.mind.AddSpell(personal_recall)
+		if(equipment && ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.delete_equipment()
+			H.equipOutfit(equipment)
+
+// the effect that happens when someone recalls to your portal
+/obj/effect/portal/permanent/one_way/recall/proc/recall_effect(mob/user)
+	return
+
+/obj/effect/proc_holder/spell/portal_recall
+	name = "Portal Recall"
+	desc = "This will teleport you back to your previously used portal. One use only."
+	clothes_req = FALSE
+	action_icon_state = "blink"
+	var/list/recall_portals = list()
+
+/obj/effect/proc_holder/spell/portal_recall/Click(mob/user = usr)
+	if(!recall_portals.len)
+		user.mind.RemoveSpell(src) // remove spell if no portals left
+	var/obj/effect/portal/permanent/one_way/recall/last_portal = recall_portals[recall_portals.len]
+	var/turf/recall_turf = get_turf(last_portal)
+	if(recall_turf)
+		if(last_portal.recall_equipment && ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.delete_equipment()
+			H.equipOutfit(last_portal.recall_equipment)
+		last_portal.recall_effect(user)
+		do_teleport(user, recall_turf, 0, no_effects = FALSE, channel = TELEPORT_CHANNEL_BLUESPACE)
+		recall_portals -= last_portal
+		if(!recall_portals.len)
+			user.mind.RemoveSpell(src) // remove spell if no portals left
+
+/obj/effect/mob_spawn/human/virtual_reality
+	name = "Network Relay"
+	desc = "A machine with flashing buttons. It seems to be some sort of teleportation pad. There doesn't seem to be any way to activate it from this side."
+	mob_name = "Virtual Reality Human"
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "qpad-idle"
+	density = FALSE
+	roundstart = FALSE
+	death = FALSE
+	uses = -1 // infinite
+	random = TRUE
+	mob_species = /datum/species/human
+	outfit = /datum/outfit/vr
+	flavour_text = "<span class='big bold'>You have connected to another stations virtual reality system</span>\n\
+	<b>You have been assigned a mission by nanotrasen to increase productivity in a station that lacks any and all teamwork.\n\
+	Your objective is to help other people by completing tasks, whether social or combatative, together, while also improving your own skills.</b>"
+	assignedrole = "Vr"
+	var/pacifist = TRUE // if the virtual human upon spawning is a pacifist
+
+/obj/effect/mob_spawn/human/virtual_reality/equip(mob/living/carbon/human/H)
+	. = ..()
+	if(pacifist)
+		H.add_trait(TRAIT_PACIFISM, TRAUMA_TRAIT)
+
+/obj/effect/portal/permanent/one_way/recall/murderdome
+	name = "Murderdome Portal"
+	desc = "Active, but only occasionally. Leads to an endless battle arena."
+	equipment = /datum/outfit/death_commando
+	recall_equipment = /datum/outfit/vr
+	id = "vr murderdome"
+	light_color = LIGHT_COLOR_FIRE
+	light_power = 1
+	light_range = 3
+
+/obj/effect/portal/permanent/one_way/recall/murderdome/teleport(atom/movable/M, force = FALSE)
+	. = ..()
+	if(. && ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.remove_trait(TRAIT_PACIFISM, TRAUMA_TRAIT)
```

Comment:
the second argument to the remove_trait proc is a list of sources, how is this valid code

More importantly, you should be adding a new source for this trait, it hasn't come from TRAUMA, it's coming from portals and should have a new define source added.

Reply:
literally every other instance of remove_trait i've seen in the code doesn't use it as a list

Reply:
the proc definition definitely takes a list, i'll have to look at the logic and see what it does

Reply:
apparently it has a check in there
```
if(!islist(sources))
        sources = list(sources)
```